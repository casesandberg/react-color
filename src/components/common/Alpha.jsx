'use strict';

var React = require('react');
var ReactCSS = require('reactcss');

var Checkboard = require('./Checkboard');

class Alpha extends ReactCSS.Component {

  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleMouseDown = this.handleMouseDown.bind(this);
    this.handleMouseUp = this.handleMouseUp.bind(this);
  }

  classes() {
    return {
      'default': {
        alpha: {
          Absolute: '0 0 0 0',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0 0 0 0',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0 0 0 0',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0 0 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    };
  }

  handleChange(e, skip) {
    !skip && e.preventDefault();
    var container = this.refs.container;
    var containerWidth = container.clientWidth;
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset);

    var a;
    if (left < 0) {
      a = 0;
    } else if (left > containerWidth) {
      a = 1;
    } else {
      a = Math.round(left * 100 / containerWidth) / 100;
    }

    if (this.props.a !== a) {
      this.props.onChange({ h: this.props.hsl.h, s: this.props.hsl.s, l: this.props.hsl.l, a: a });
    }
  }

  handleMouseDown(e) {
    this.handleChange(e, true);
    window.addEventListener('mousemove', this.handleChange);
    window.addEventListener('mouseup', this.handleMouseUp);
  }

  handleMouseUp() {
    window.removeEventListener('mousemove', this.handleChange);
    window.removeEventListener('mouseup', this.handleMouseUp);
  }

  render() {
    var pointer = <div is="slider" />;

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />;
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown } onTouchMove={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    );
  }
}

module.exports = Alpha;
