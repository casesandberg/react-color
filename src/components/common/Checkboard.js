'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

let _checkboardCache = {}

function renderCheckboard(c1: string, c2: string, size: number): any {
  if (typeof document == 'undefined') return null // Dont Render On Server
  var canvas: any = document.createElement('canvas')
  canvas.width = canvas.height = size * 2
  var ctx = canvas.getContext('2d')
  if (!ctx) return null // If no context can be found, return early.
  ctx.fillStyle = c1
  ctx.fillRect(0, 0, canvas.width, canvas.height)
  ctx.fillStyle = c2
  ctx.fillRect(0, 0, size, size)
  ctx.translate(size, size)
  ctx.fillRect(0, 0, size, size)
  return canvas.toDataURL()
}

function getCheckboard(c1: string, c2: string, size: number): any {
  var key = c1 + ',' + c2 + ',' + size

  if (_checkboardCache[key]) {
    return _checkboardCache[key]
  } else {
    var checkboard = renderCheckboard(c1, c2, size)
    _checkboardCache[key] = checkboard
    return checkboard
  }
}

export class Checkboard extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    var background = getCheckboard(this.props.white, this.props.grey, this.props.size)
    return {
      'default': {
        grid: {
          Absolute: '0px 0px 0px 0px',
          background: 'url(' + background + ') center left',
        },
      },
    }
  }

  render(): any {
    return (
      <div is="grid" ref="grid"></div>
    )
  }

}

Checkboard.defaultProps = {
  size: 8,
  white: '#fff',
  grey: '#e6e6e6',
}

export default Checkboard
