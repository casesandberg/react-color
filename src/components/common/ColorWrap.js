'use strict' /* @flow */

import React from 'react'
import merge from 'merge'
import isPlainObject from 'lodash/isplainobject'
import debounce from 'lodash/debounce'
import color from '../../helpers/color'
import shallowCompare from 'react-addons-shallow-compare'

export const ColorWrap = (Picker) => {
  class ColorPicker extends React.Component {
    shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1]);

    constructor(props: any) {
      super()

      this.state = merge(color.toState(props.color, 0), {
        visible: props.display,
      })

      this.debounce = debounce(function (fn: any, data: any) {
        fn(data)
      }, 100)
    }

    handleChange = (data: any) => {
      data = color.simpleCheckForValidColor(data)
      if (data) {
        var colors = color.toState(data, data.h || this.state.oldHue)
        this.setState(colors)
        this.props.onChangeComplete && this.debounce(this.props.onChangeComplete, colors)
        this.props.onChange && this.props.onChange(colors)
      }
    }

    componentWillReceiveProps(nextProps: any) {
      this.setState(merge(color.toState(nextProps.color, this.state.oldHue), {
        visible: nextProps.display,
      }))
    }

    render(): any {
      return <Picker {...this.props} {...this.state} onChange={ this.handleChange } />
    }
  }

  ColorPicker.defaultProps = {
    color: {
      h: 250,
      s: .50,
      l: .20,
      a: 1,
    },
  }

  return ColorPicker
}

export default ColorWrap
