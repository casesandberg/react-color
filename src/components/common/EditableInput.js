import React, { Component, PureComponent } from 'react'
import reactCSS from 'reactcss'

const DEFAULT_ARROW_OFFSET = 1

const UP_KEY_CODE = 38
const DOWN_KEY_CODE = 40
const VALID_KEY_CODES = [
  UP_KEY_CODE,
  DOWN_KEY_CODE
]
const isValidKeyCode = keyCode => VALID_KEY_CODES.indexOf(keyCode) > -1
const getNumberValue = value => Number(String(value).replace(/%/g, ''))

let idCounter = 1

export class EditableInput extends (PureComponent || Component) {
  constructor(props) {
    super()

    this.state = {
      value: String(props.value).toUpperCase(),
      blurValue: String(props.value).toUpperCase(),
    }

    this.inputId = `rc-editable-input-${idCounter++}`
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      this.props.value !== this.state.value &&
      (prevProps.value !== this.props.value || prevState.value !== this.state.value)
    ) {
      if (this.input === document.activeElement) {
        this.setState({ blurValue: String(this.props.value).toUpperCase() })
      } else {
        this.setState({ value: String(this.props.value).toUpperCase(), blurValue: !this.state.blurValue && String(this.props.value).toUpperCase() })
      }
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  getValueObjectWithLabel(value) {
    return {
      [this.props.label]: value
    }
  }

  handleBlur = () => {
    if (this.state.blurValue) {
      this.setState({ value: this.state.blurValue, blurValue: null })
    }
  }

  handleChange = (e) => {
    this.setUpdatedValue(e.target.value, e)
  }

  getArrowOffset() {
    return this.props.arrowOffset || DEFAULT_ARROW_OFFSET
  }

  handleKeyDown = (e) => {
    // In case `e.target.value` is a percentage remove the `%` character
    // and update accordingly with a percentage
    // https://github.com/casesandberg/react-color/issues/383
    const value = getNumberValue(e.target.value)
    if (!isNaN(value) && isValidKeyCode(e.keyCode)) {
      const offset = this.getArrowOffset()
      const updatedValue = e.keyCode === UP_KEY_CODE ? value + offset : value - offset

      this.setUpdatedValue(updatedValue, e)
    }
  }

  setUpdatedValue(value, e) {
    const onChangeValue = this.props.label ? this.getValueObjectWithLabel(value) : value
    this.props.onChange && this.props.onChange(onChangeValue, e)

    this.setState({ value })
  }

  handleDrag = (e) => {
    if (this.props.dragLabel) {
      const newValue = Math.round(this.props.value + e.movementX)
      if (newValue >= 0 && newValue <= this.props.dragMax) {
        this.props.onChange && this.props.onChange(this.getValueObjectWithLabel(newValue), e)
      }
    }
  }

  handleMouseDown = (e) => {
    if (this.props.dragLabel) {
      e.preventDefault()
      this.handleDrag(e)
      window.addEventListener('mousemove', this.handleDrag)
      window.addEventListener('mouseup', this.handleMouseUp)
    }
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleDrag)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render() {
    const styles = reactCSS({
      'default': {
        wrap: {
          position: 'relative',
        },
      },
      'user-override': {
        wrap: this.props.style && this.props.style.wrap ? this.props.style.wrap : {},
        input: this.props.style && this.props.style.input ? this.props.style.input : {},
        label: this.props.style && this.props.style.label ? this.props.style.label : {},
      },
      'dragLabel-true': {
        label: {
          cursor: 'ew-resize',
        },
      },
    }, {
      'user-override': true,
    }, this.props)

    return (
      <div style={ styles.wrap }>
        <input
          id={ this.inputId }
          style={ styles.input }
          ref={ input => this.input = input }
          value={ this.state.value }
          onKeyDown={ this.handleKeyDown }
          onChange={ this.handleChange }
          onBlur={ this.handleBlur }
          placeholder={ this.props.placeholder }
          spellCheck="false"
        />
        { this.props.label && !this.props.hideLabel ? (
          <label
            htmlFor={ this.inputId }
            style={ styles.label }
            onMouseDown={ this.handleMouseDown }
          >
            { this.props.label }
          </label>
        ) : null }
      </div>
    )
  }
}

export default EditableInput
