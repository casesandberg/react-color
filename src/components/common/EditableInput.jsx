'use strict';

var React = require('react');
var ReactCSS = require('reactcss');

class EditableInput extends ReactCSS.Component {

  constructor(props) {
    super();

    this.state = {
      value: String(props.value).toUpperCase(),
      blurValue: String(props.value).toUpperCase(),
    },

    this.handleChange = this.handleChange.bind(this);
    this.handleDrag = this.handleDrag.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.handleMouseDown = this.handleMouseDown.bind(this);
    this.handleMouseUp = this.handleMouseUp.bind(this);
  }

  classes() {
    return {
      'user-override': {
        wrap: this.props.style && this.props.style.wrap ? this.props.style.wrap : {},
        input: this.props.style && this.props.style.input ? this.props.style.input : {},
        label: this.props.style && this.props.style.label ? this.props.style.label : {},
      },
      'dragLabel-true': {
        label: {
          cursor: 'ew-resize',
        },
      },
    };
  }

  styles() {
    return this.css({
      'user-override': true,
    });
  }

  componentWillReceiveProps(nextProps) {
    var input = this.refs.input;
    if (nextProps.value !== this.state.value) {
      if (input === document.activeElement) {
        this.setState({ blurValue: String(nextProps.value).toUpperCase() });
      } else {
        this.setState({ value: String(nextProps.value).toUpperCase() });
      }
    }
  }

  handleBlur() {
    if (this.state.blurValue) {
      this.setState({ value: this.state.blurValue, blurValue: null });
    }
  }

  handleChange(e) {
    if (this.props.label !== null) {
      var obj = {};
      obj[this.props.label] = e.target.value;
      this.props.onChange(obj);
    } else {
      this.props.onChange(e.target.value);
    }

    this.setState({ value: e.target.value });
  }

  handleKeyDown(e) {
    var number = Number(e.target.value);
    if (number) {
      var amount = this.props.arrowOffset || 1;

      // Up
      if (e.keyCode === 38) {
        if (this.props.label !== null) {
          var obj = {};
          obj[this.props.label] = number + amount;
          this.props.onChange(obj);
        } else {
          this.props.onChange(number + amount);
        }

        this.setState({ value: number + amount });
      }

      // Down
      if (e.keyCode === 40) {
        if (this.props.label !== null) {
          var obj = {};
          obj[this.props.label] = number - amount;
          this.props.onChange(obj);
        } else {
          this.props.onChange(number - amount);
        }

        this.setState({ value: number - amount });
      }

    }
  }

  handleDrag(e) {
    if (this.props.dragLabel) {
      var newValue = Math.round(this.props.value + e.movementX);
      if (newValue >= 0 && newValue <= this.props.dragMax) {
        var obj = {};
        obj[this.props.label] = newValue;
        this.props.onChange(obj);
      }
    }
  }

  handleMouseDown(e) {
    if (this.props.dragLabel) {
      e.preventDefault();
      this.handleDrag(e);
      window.addEventListener('mousemove', this.handleDrag);
      window.addEventListener('mouseup', this.handleMouseUp);
    }
  }

  handleMouseUp() {
    window.removeEventListener('mousemove', this.handleDrag);
    window.removeEventListener('mouseup', this.handleMouseUp);
  }

  render() {
    var label;
    if (this.props.label) {
      label = <span is="label" ref="label" onMouseDown={ this.handleMouseDown }>{ this.props.label }</span>;
    }

    return (
      <div is="wrap" ref="container">
        <input is="input" ref="input" value={ this.state.value } onKeyDown={ this.handleKeyDown } onChange={ this.handleChange } onBlur={ this.handleBlur }/>
        { label }
      </div>
    );
  }
}

module.exports = EditableInput;
