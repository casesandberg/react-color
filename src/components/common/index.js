export { default as Alpha } from './Alpha'
export { default as Checkboard } from './Checkboard'
export { default as EditableInput } from './EditableInput'
export { default as Hue } from './Hue'
export { default as Raised } from './Raised'
export { default as Saturation } from './Saturation'
export { default as ColorWrap } from './ColorWrap'
export { default as Swatch } from './Swatch'
