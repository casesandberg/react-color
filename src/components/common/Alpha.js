'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

import Checkboard from './Checkboard'

export class Alpha extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        alpha: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0px 0px 0px 0px',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0px 0px 0px 0px',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0px 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0px 0px 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  handleChange = (e: any, skip: boolean) => {
    !skip && e.preventDefault()
    var container = this.refs.container
    var containerWidth = container.clientWidth
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset)

    var a
    if (left < 0) {
      a = 0
    } else if (left > containerWidth) {
      a = 1
    } else {
      a = Math.round(left * 100 / containerWidth) / 100
    }

    if (this.props.a !== a) {
      this.props.onChange({
        h: this.props.hsl.h,
        s: this.props.hsl.s,
        l: this.props.hsl.l,
        a: a,
        source: 'rgb',
      })
    }
  }

  handleMouseDown = (e: any) => {
    this.handleChange(e, true)
    window.addEventListener('mousemove', this.handleChange)
    window.addEventListener('mouseup', this.handleMouseUp)
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleChange)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render(): any {
    var pointer = <div is="slider" />

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown }
            onTouchMove={ this.handleChange }
            onTouchStart={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    )
  }
}

export default Alpha
