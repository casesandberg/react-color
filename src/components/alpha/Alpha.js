import React from 'react'
import reactCSS from 'reactcss'

import { ColorWrap, Alpha } from '../common'
import AlphaPointer from './AlphaPointer'

export const AlphaPicker = ({ rgb, hsl, width, height, onChange, direction, style,
  renderers, pointer, className = '' }) => {
  const styles = reactCSS({
    'default': {
      picker: {
        position: 'relative',
        width,
        height,
      },
      alpha: {
        radius: '2px',
        style,
      },
    },
  })

  return (
    <div style={ styles.picker } className={ `alpha-picker ${ className }` }>
      <Alpha
        { ...styles.alpha }
        rgb={ rgb }
        hsl={ hsl }
        pointer={ pointer }
        renderers={ renderers }
        onChange={ onChange }
        direction={ direction }
      />
    </div>
  )
}

AlphaPicker.defaultProps = {
  width: '316px',
  height: '16px',
  direction: 'horizontal',
  pointer: AlphaPointer,
}

export default ColorWrap(AlphaPicker)
