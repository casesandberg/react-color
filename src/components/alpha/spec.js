/* global test, expect, jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import color, { red } from '../../helpers/color'
// import canvas from 'canvas'

import Alpha from './Alpha'
import { Alpha as CommonAlpha } from '../common'
import AlphaPointer from './AlphaPointer'

test('Alpha renders correctly', () => {
  const tree = renderer.create(
    <Alpha { ...red } />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

// test('Alpha renders on server correctly', () => {
//   const tree = renderer.create(
//     <Alpha renderers={{ canvas }} { ...red } />
//   ).toJSON()
//   expect(tree).toMatchSnapshot()
// })

test('Alpha onChange events correctly', () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Alpha { ...red } width={ 20 } height={ 200 } onChange={ changeSpy } />,
  )
  expect(changeSpy).toHaveBeenCalledTimes(0)
  const alphaCommon = tree.find(CommonAlpha)
  alphaCommon.at(0).childAt(2).simulate('mouseDown', {
    pageX: 100,
    pageY: 10,
  })
  expect(changeSpy).toHaveBeenCalled()
})

test('Alpha renders vertically', () => {
  const tree = renderer.create(
    <Alpha { ...red } width={ 20 } height={ 200 } direction="vertical" />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('AlphaPointer renders correctly', () => {
  const tree = renderer.create(
    <AlphaPointer />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
