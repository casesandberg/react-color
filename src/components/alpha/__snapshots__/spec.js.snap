// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alpha renders correctly 1`] = `
<div
  className="alpha-picker "
  style={
    Object {
      "height": "16px",
      "position": "relative",
      "width": "316px",
    }
  }
>
  <div
    style={
      Object {
        "MozBorderRadius": "2px",
        "OBorderRadius": "2px",
        "WebkitBorderRadius": "2px",
        "borderRadius": "2px",
        "bottom": "0px",
        "left": "0px",
        "msBorderRadius": "2px",
        "position": "absolute",
        "right": "0px",
        "top": "0px",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": "0px",
          "left": "0px",
          "overflow": "hidden",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <div
        style={
          Object {
            "MozBorderRadius": undefined,
            "MozBoxShadow": undefined,
            "OBorderRadius": undefined,
            "OBoxShadow": undefined,
            "WebkitBorderRadius": undefined,
            "WebkitBoxShadow": undefined,
            "background": "url(null) center left",
            "borderRadius": undefined,
            "bottom": "0px",
            "boxShadow": undefined,
            "left": "0px",
            "msBorderRadius": undefined,
            "msBoxShadow": undefined,
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "MozBoxShadow": undefined,
          "OBorderRadius": "2px",
          "OBoxShadow": undefined,
          "WebkitBorderRadius": "2px",
          "WebkitBoxShadow": undefined,
          "background": "linear-gradient(to right, rgba(34,25,77, 0) 0%,
                   rgba(34,25,77, 1) 100%)",
          "borderRadius": "2px",
          "bottom": "0px",
          "boxShadow": undefined,
          "left": "0px",
          "msBorderRadius": "2px",
          "msBoxShadow": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "height": "100%",
          "margin": "0 3px",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "left": "100%",
            "position": "absolute",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "50%",
              "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "MozTransform": "translate(-9px, -1px)",
              "OBorderRadius": "50%",
              "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "OTransform": "translate(-9px, -1px)",
              "WebkitBorderRadius": "50%",
              "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "WebkitTransform": "translate(-9px, -1px)",
              "backgroundColor": "rgb(248, 248, 248)",
              "borderRadius": "50%",
              "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "height": "18px",
              "msBorderRadius": "50%",
              "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "msTransform": "translate(-9px, -1px)",
              "transform": "translate(-9px, -1px)",
              "width": "18px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Alpha renders vertically 1`] = `
<div
  className="alpha-picker "
  style={
    Object {
      "height": 200,
      "position": "relative",
      "width": 20,
    }
  }
>
  <div
    style={
      Object {
        "MozBorderRadius": "2px",
        "OBorderRadius": "2px",
        "WebkitBorderRadius": "2px",
        "borderRadius": "2px",
        "bottom": "0px",
        "left": "0px",
        "msBorderRadius": "2px",
        "position": "absolute",
        "right": "0px",
        "top": "0px",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": "0px",
          "left": "0px",
          "overflow": "hidden",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    >
      <div
        style={
          Object {
            "MozBorderRadius": undefined,
            "MozBoxShadow": undefined,
            "OBorderRadius": undefined,
            "OBoxShadow": undefined,
            "WebkitBorderRadius": undefined,
            "WebkitBoxShadow": undefined,
            "background": "url(null) center left",
            "borderRadius": undefined,
            "bottom": "0px",
            "boxShadow": undefined,
            "left": "0px",
            "msBorderRadius": undefined,
            "msBoxShadow": undefined,
            "position": "absolute",
            "right": "0px",
            "top": "0px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "MozBorderRadius": "2px",
          "MozBoxShadow": undefined,
          "OBorderRadius": "2px",
          "OBoxShadow": undefined,
          "WebkitBorderRadius": "2px",
          "WebkitBoxShadow": undefined,
          "background": "linear-gradient(to bottom, rgba(34,25,77, 0) 0%,
                   rgba(34,25,77, 1) 100%)",
          "borderRadius": "2px",
          "bottom": "0px",
          "boxShadow": undefined,
          "left": "0px",
          "msBorderRadius": "2px",
          "msBoxShadow": undefined,
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      onMouseDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "height": "100%",
          "margin": "0 3px",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "top": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "MozBorderRadius": "50%",
              "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "MozTransform": "translate(-3px, -9px)",
              "OBorderRadius": "50%",
              "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "OTransform": "translate(-3px, -9px)",
              "WebkitBorderRadius": "50%",
              "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "WebkitTransform": "translate(-3px, -9px)",
              "backgroundColor": "rgb(248, 248, 248)",
              "borderRadius": "50%",
              "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "height": "18px",
              "msBorderRadius": "50%",
              "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
              "msTransform": "translate(-3px, -9px)",
              "transform": "translate(-3px, -9px)",
              "width": "18px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AlphaPointer renders correctly 1`] = `
<div
  style={
    Object {
      "MozBorderRadius": "50%",
      "MozBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "MozTransform": "translate(-9px, -1px)",
      "OBorderRadius": "50%",
      "OBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "OTransform": "translate(-9px, -1px)",
      "WebkitBorderRadius": "50%",
      "WebkitBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "WebkitTransform": "translate(-9px, -1px)",
      "backgroundColor": "rgb(248, 248, 248)",
      "borderRadius": "50%",
      "boxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "height": "18px",
      "msBorderRadius": "50%",
      "msBoxShadow": "0 1px 4px 0 rgba(0, 0, 0, 0.37)",
      "msTransform": "translate(-9px, -1px)",
      "transform": "translate(-9px, -1px)",
      "width": "18px",
    }
  }
/>
`;
